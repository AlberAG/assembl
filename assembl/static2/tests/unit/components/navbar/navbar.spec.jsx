import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {
  AssemblNavbar,
  sectionMapper,
  mapSectionToElement,
  __RewireAPI__ as NavbarRewireAPI
} from '../../../../js/app/components/navbar/navbar';
import rewireFunction from '../../../helpers/rewireFunction';

describe('sectionMapper function', () => {
  it('should return a function', () => {
    const ret = sectionMapper();
    expect(typeof ret).toBe('function');
  });
  it('should return a function that calls mapSectionToElement', () => {
    const mock = jest.fn();
    rewireFunction(NavbarRewireAPI, 'mapSectionToElement', mock, () => {
      const mapSection = sectionMapper();
      mapSection();
      expect(mock).toHaveBeenCalled();
    });
  });
  it('should return a function that returns mapSectionToElement\'s return value', () => {
    const mockReturn = { someProp: 'bar' };
    const mock = jest.fn(() => {
      return mockReturn;
    });
    rewireFunction(NavbarRewireAPI, 'mapSectionToElement', mock, () => {
      const mapSection = sectionMapper();
      const ret = mapSection();
      expect(ret).toBe(mockReturn);
    });
  });
  it('should return a function that calls mapSectionToElement with the function\'s argument and the sectionMapper\'s options', () => {
    const options = { someOption: true };
    const section = { someSectionProp: 'foo' };
    const mock = jest.fn();
    rewireFunction(NavbarRewireAPI, 'mapSectionToElement', mock, () => {
      const mapSection = sectionMapper(options);
      mapSection(section);
      expect(mock).toHaveBeenCalledWith(section, options);
    });
  });
});

describe('mapSectionToElement function', () => {
  it('should return a SectionLink element that matches the snapshot', () => {
    const ret = mapSectionToElement({ sectionType: 'DEBATE', title: 'fake_title' }, { slug: 'fake_slug', phase: 'fake_phase' });
    expect(ret).toMatchSnapshot();
  });
});

describe('AssemblNavbar component', () => {
  it('should render a FlatNavbar with a big screenWidth', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<AssemblNavbar screenWidth={2000} debate={{ debateData: { timeline: [] } }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('should render a BurgerNavbar and a hidden FlatNavbar with a tiny screenWidth', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<AssemblNavbar screenWidth={10} debate={{ debateData: { timeline: [] } }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});