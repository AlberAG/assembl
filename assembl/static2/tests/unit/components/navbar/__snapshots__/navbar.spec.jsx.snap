// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssemblNavbar component should render a BurgerNavbar and a hidden FlatNavbar with a tiny screenWidth 1`] = `
<div
  className="background-light"
>
  <Uncontrolled(Navbar)
    bsStyle="default"
    fixedTop={true}
    fluid={true}
  >
    <div
      className="nav-bar max-container"
      id="navbar"
    >
      <BurgerNavbar
        elements={Array []}
        helpUrl={undefined}
        location={undefined}
        logoSrc={undefined}
        slug={undefined}
      />
      <FlatNavbar
        elements={Array []}
        helpUrl={undefined}
        location={undefined}
        logoSrc={undefined}
        setWidth={[Function]}
        slug={undefined}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "top": "-200px",
          }
        }
      />
    </div>
  </Uncontrolled(Navbar)>
</div>
`;

exports[`AssemblNavbar component should render a FlatNavbar with a big screenWidth 1`] = `
<div
  className="background-light"
>
  <Uncontrolled(Navbar)
    bsStyle="default"
    fixedTop={true}
    fluid={true}
  >
    <div
      className="nav-bar max-container"
      id="navbar"
    >
      <FlatNavbar
        elements={Array []}
        helpUrl={undefined}
        location={undefined}
        logoSrc={undefined}
        setWidth={[Function]}
        slug={undefined}
        style={Object {}}
      />
    </div>
  </Uncontrolled(Navbar)>
</div>
`;

exports[`mapSectionToElement function should return a SectionLink element that matches the snapshot 1`] = `
<SectionLink
  title="fake_title"
  to="/fake_slug/debate/fake_phase"
/>
`;
